### PR TITLE
i18n(app.yml): new language zh_TW

### PR DESCRIPTION
**Pull request by olivertzeng**
translated topgrade to zh_TW

## Standards checklist

- [x] The PR title is descriptive.
- [x] I have read `CONTRIBUTING.md`
- [x] *Optional:* I have tested the code myself
 
## For new steps

- [ ] *Optional:* Topgrade skips this step where needed
- [ ] *Optional:* The `--dry-run` option works with this step
- [ ] *Optional:* The `--yes` option works with this step if it is supported by 
  the underlying command
#2 

Notes: the tests were successful but the locales doesn't show somehow

### CODE REVIEW

File: locales/app.yml [hunk 1]
```diff
@@ -1,307 +1,464 @@
...
+  zh_TW: "已更新所有擴充功能"
 "Would self-update":
   en: "Would self-update"
+  zh_TW: "將自動更新"
 "Pulling":
   en: "Pulling"
+  zh_TW: "正在拉取"
 "No Breaking changes":
   en: "No Breaking changes"
+  en: "無重大更改"
```
Comment (SteveLauC) [inline locales/app.yml:26]:
```suggestion
  zh_TW: "無重大更改"
```

Caught by our i18n checker: https://github.com/topgrade-rs/topgrade/actions/runs/11209992435/job/31156103528?pr=931

Comment (olivertzeng) [reply locales/app.yml:26]: Thanks for the heads up

**Review by SteveLauC — APPROVED** [review]
Thanks for the PR, left some comments.

**Comment by SteveLauC** [comment]
CI is not running because we have conflicts.

File: locales/app.yml [hunk 2]
```diff
@@ -1,307 +1,464 @@
...
+  zh_TW: "Nix-darwin 在 macOS 上必須使用 darwin-rebuild 更新"
 "`nix upgrade-nix` can only be used on macOS or non-NixOS Linux":
   en: "`nix upgrade-nix` can only be used on macOS or non-NixOS Linux"
+  zh_TW: "`nix upgrade-nix` 僅能在 macOS 或非 NixOS 的 Linux 上使用"
 "`nix upgrade-nix` cannot be run when Nix is installed in a profile":
   en: "`nix upgrade-nix` cannot be run when Nix is installed in a profile"
+  zh_TW: "`nix upgrade-nix` 無法在已安裝 Nix 使用者環境的系統上使用"
 "Nix (self-upgrade)":
   en: "Nix (self-upgrade)"
+  zh_TW: "Nix（自動更新）"
```
[the remark this answers is not in the document]
Comment (olivertzeng) [reply locales/app.yml:278]: Whoops

Will fix that(sometimes I make mistakes sorry :()

File: locales/app.yml [hunk 3]
```diff
@@ -1,307 +1,464 @@
...
+  zh_TW: "Deno 安裝在 .deno 資料夾外"
 "The Ultimate vimrc":
   en: "The Ultimate vimrc"
+  zh_TW: "終極 vimrc（The Ultimate vimrc）"
 "vim binary might be actually nvim":
   en: "vim binary might be actually nvim"
+  zh_TW: "vim 執行檔可能為 nvim"
 "`{process}` failed: {exit_satus}":
   en: "`%{process}` failed: %{exit_satus}"
+  zh_TW: "`%{process}` 失敗：%{exit_satus}"
```
[the remark this answers is not in the document]
Comment (olivertzeng) [reply locales/app.yml:383]: Will do
I'll merge them asap after I finish school

**Comment by olivertzeng** [comment]
@SteveLauC I suggest having po and mo files or such that separates different languages

That way translating will be much much easier

**Comment by olivertzeng** [comment]
> CI is not running because we have conflicts.

Will solve this tonight

**Comment by SteveLauC** [comment]
> I suggest having po and mo files or such that separates different languages

Interesting, TIL, I agree, though I am not sure if [the i18n library](https://docs.rs/rust-i18n/latest/rust_i18n/) used by Topgrade is capable of coping with this. 🤔

File: locales/app.yml [hunk 4]
```diff
@@ -1,307 +1,464 @@
...
   en: "Up-to-date"
+  zh_TW: "已為最新版本"
 "Self update":
   en: "Self update"
-  
+  zh_TW: "自行更新"
+
 # The following 2 strings are used in the same sentence
+# i.e. *Only* updated repositories will be shown...
+# Note that the text *Only* is highlighted in green
```
[the remark this answers is not in the document]
Comment (olivertzeng) [reply locales/app.yml:72]: Leaving contexts to translations is important to me
same for the others

**Review by SteveLauC — APPROVED** [review]
LGTM, thanks for the patch!

**Comment by Mastermindaxe** [comment]
> > I suggest having po and mo files or such that separates different languages
> 
> Interesting, TIL, I agree, though I am not sure if [the i18n library](https://docs.rs/rust-i18n/latest/rust_i18n/) used by Topgrade is capable of coping with this. 🤔

@SteveLauC 
We could have different files for each language. But is it really better to have them separated? When everything is in one file you can see immediately if there is a key missing for a language you're implementing. On the other hand you'd probably just copy over the english translation file and fill that out step-by-step 🤔 I'd be okay with both. The framework certainly supports it if I recall correctly

**Comment by SteveLauC** [comment]
>  When everything is in one file you can see immediately if there is a key missing for a language you're implementing. On the other hand you'd probably just copy over the english translation file and fill that out step-by-step

Yeah, putting translations together will make adding new language support and checking easier. Let's keep the current 
 approach.

**Pull request merged**